src/rdds/lib/hdf5/hd5_data_generator.py

```diff
@@ -163,7 +165,7 @@ def _expand_categorical_label_1d_to_2d(label: float) -> np.array:
         if not 0.0 <= label <= 1.0:
```

Comment:
this check could be abstracted away to something like:

```suggestion
        if not is_label_valid(label):
```


then 
```Python
def is_label_valid(label):
return 0.0 <= label <= 1.0
```

Not strictly necessary but it makes the code more readable, plus you can test the is_valid_function with unittests and make sure it works like you want to.